### ISSUE TITLE
Need help on including pybind11 in my cmake project using `find_package()`

### ISSUE THREAD

**Issue by chongkong** [pybind/pybind11#1588]
I'm trying to use pybind11 with CMake's `find_package()` as described in the [doc](https://github.com/pybind/pybind11/blob/master/docs/compiling.rst#find_package-vs-add_subdirectory) and [code](https://github.com/pybind/pybind11/blob/master/tools/pybind11Config.cmake.in). Unfortunately, things doesn't work if you just follow the document. After some trials and errors I succeeded at building my project, but what I've done seems to be far from the right way.

Here's what I've done:

1. **Build the cloned pybind11 in `build/` directory**. After this step I have `pybind11Config.cmake` file ready in `build/` directory.
``` shell
$ cd build   # at pybind11 project root
$ cmake ..
$ make
```
2. **Use `find_package()` with `PATH_SUFFIXES` option to properly locate `pybind11`**.
``` cmake
find_package(pybind11 REQUIRED
             PATHS /absolute/path/to/pybind11
             PATH_SUFFIXES build)
```
3. There are some files `pybind11Config.cmake` includes, but doesn't exists in `build/` directory. So I searched and **manually copied the files into `build/` directory**. Missing files includes:
    - `pybind11Tools.cmake` is in a `tools/` directory.
    - `FindPythonLibsNew.cmake`also in a `tools/` directory.
    - `pybind11Targets.cmake` is generated in the `build/CMakeFiles/Export/share/pybind11` directory.

These steps are not only cumbersome but also undocumented. Maybe I'm missing something critical, or the document is staled. 

So what I'd like to ask you is, what is the proper way to setup pybind11 in external CMake project without manually copying the `*.cmake` files?

**Comment by LeslieGerman** [comment]
According to [Changelog](https://pybind11.readthedocs.io/en/master/changelog.html#v2-4-0-sep-19-2019), this has been addressed/fixed in v2.4.0 (Sep 19, 2019).
(I haven't tried.)

**Comment by bstaletic** [comment]
I've given this a test and at least on 2.5.0 it definitely works.
